### ISSUE TITLE
Better support for zsh

### ISSUE THREAD

**Issue by jansorg** [BashSupport-Pro/bashsupport-pro#23]
Improve support for zsh support.
JetBrains Shell is handling .zsh files, but identically to Bash files. 

Things, which BashSupport Pro could do:
- add support for additional/different syntax elements of zsh
- Support the zsh shebang lines
- Support the zsh debugger, https://github.com/rocky/zshdb
- Support commonly used tools of the zsh world. Is there something similar like shellcheck or shellfmt?
- Show zsh docs for built-in commands and keywords, instead of bash help content

**Comment by jordantrizz** [comment]
ZSH code completion would also be awesome.

**Comment by freddy33** [comment]
Yes. Kind of blocked now since I added `${(%):-%x}` in my zsh script Bash support pro cannot analyze any vars after :(

**Comment by jansorg** [comment]
@freddy33 Could you post a snippet where this happens? The parsing should recover and start to parse commands again after the unsupported zsh statement.

**Comment by jansorg** [comment]
I'm working on Zsh support now, which will be part of version 4.0. This is going to take some time, because it's a lot of work. I'm planning to release preview versions as soon as Zsh support is somewhat stable.

**Comment by jansorg** [comment]
Preview build 4.0.0-beta9 is available with zshdb integration for local and remote debugging, completion and a few more features. Changelog and download for testing at https://plugins.jetbrains.com/plugin/13841-bashsupport-pro/eap-bashsupport-pro-4-0
I'd appreciate if you shared your feedback. Thank you!

**Comment by jansorg** [comment]
4.0 beta11 is now able to parse Zsh‘s parameter completion flags.

**Comment by jansorg** [comment]
4.o with Zsh support is available now, https://www.bashsupport.com/manual/zsh/ lists features and known limitations

**Comment by joshuataylor** [comment]
Just want to add, that the Bash, and now zsh debugging through this plugin is amazing. Been using it to figure out some weirdness going on with my .zshrc :-).

**Comment by jansorg** [comment]
5.0 has been released with a built-in formatter for Zsh
